src/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -218,6 +218,89 @@ public void ProcessStart_UseShellExecute_OnWindows_DoesNotThrow(bool isFolder)
             }
         }
 
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsServerCore),
+            nameof(PlatformDetection.IsNotWindowsNanoServer), nameof(PlatformDetection.IsNotWindowsIoTCore))]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Uap, "not supported on UAP")]
+        [InlineData(true), InlineData(false)]
+        public void ProcessStart_UseShellExecute_Executes(bool filenameAsUrl)
+        {
+            string filename = GetTestFilePath();
+            if (PlatformDetection.IsWindows)
+            {
+                filename += ".bat";
+            }
+            WriteScriptFile(filename, returnValue: 42);
+
+            if (filenameAsUrl)
+            {
+                filename = new Uri(filename).ToString();
+            }
+
+            using (var process = Process.Start(new ProcessStartInfo { UseShellExecute = true, FileName = filename }))
+            {
+                process.WaitForExit();
+                Assert.Equal(42, process.ExitCode);
+            }
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsServerCore),
+            nameof(PlatformDetection.IsNotWindowsNanoServer), nameof(PlatformDetection.IsNotWindowsIoTCore))]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Uap, "not supported on UAP")]
+        public void ProcessStart_UseShellExecute_ExecuteOrder()
+        {
+            // Create a directory that we will use as PATH
+            string path = GetTestFileName();
+            Directory.CreateDirectory(path);
+
+            RemoteInvokeOptions options = new RemoteInvokeOptions();
+            options.StartInfo.EnvironmentVariables["PATH"] = path;
+            RemoteInvoke(pathDirectory =>
+            {
+                // Create two identically named scripts, one in the working directory and one on PATH.
+                string scriptFilename = $"script_{nameof(ProcessStart_UseShellExecute_ExecuteOrder)}";
```

Comment:
this should be something along the lines of `System.IO.Path.GetTempFileName()`

Reply:
Better still, use the `FileCleanupTestBase` base class and call `GetTestFilePath()`, the cleanup happens for free.

Reply:
GetTestFilePath() value would have to be passed in to the RemoteInvoke.

Reply:
My bad, I didn't read the code properly. It's already doing `GetTestFileName()` and creating that directory. The one thing I would change is to not assume that `Directory.GetCurrentDirectory()` gives a writeable path. I would create another directory (eg using the result of GetTestFileName(someindex)) and temporarily change the working directory to that.